env/plugin-commands-env/src/node.ts

```diff
@@ -36,6 +37,27 @@ export type NvmNodeCommandOptions = Pick<Config,
   remote?: boolean
 }
 
+export type ConfigWithExtraBinPaths = NvmNodeCommandOptions & Partial<Pick<Config, 'extraBinPaths'>>
+
+const nodeFetchPromises: Record<string, Promise<string>> = {}
+
+export async function prepareExecutionEnv (config: NvmNodeCommandOptions, { extraBinPaths, executionEnv }: PrepareExecutionEnvOptions): Promise<PrepareExecutionEnvResult> {
+  if (!executionEnv?.nodeVersion) return { extraBinPaths: extraBinPaths ?? [] }
+
+  let nodePathPromise = nodeFetchPromises[executionEnv.nodeVersion]
+  if (!nodePathPromise) {
+    nodePathPromise = getNodeBinDir({
+      ...config,
+      useNodeVersion: executionEnv.nodeVersion,
+    })
+    nodeFetchPromises[executionEnv.nodeVersion] = nodePathPromise
+  }
+
+  return {
+    extraBinPaths: [await nodePathPromise, ...extraBinPaths ?? []],
```

Comment:
it should be added to the end

Reply:
> it should be added to the end

It would allow `extraBinPaths` to override `node`. Do you still want this?

Reply:
isn't that the whole point? To override node with the version specified in the setting.

Reply:
It would allow the tail of `extraBinPaths` to override the `node` provided by the `nodePathPromise`. The tail of `extraBinPaths` may or may not include `node` from `use-node-version` of `.npmrc`. It would defeat the whole point.

Reply:
Yes, you are correct. The directories at the beginning have bigger priority. It is correct as it is now.

Reply:
I have added a test that would fail should `extraBinPaths` was handled incorrectly (leading to `use-node-version` overrides `executionEnv`): https://github.com/pnpm/pnpm/pull/8277/commits/2721d44f741c99ea11911e523064abc594e5c22c

You may revert if you find it unnecessary.